fabrics/helpers/distances.py

```diff
@@ -87,4 +173,136 @@ def capsule_to_sphere(
         sphere_center, capsule_centers[0], capsule_centers[1]
     )
-    return distance_line_center - capsule_radius - sphere_radius
+    return ca.fmax(distance_line_center - capsule_radius - sphere_radius, 0.0)
+
+
+def cuboid_to_point_half_distances(
+    rectangle_center: ca.SX,
+    rectangle_size: ca.SX,
+    point: ca.SX,
+) -> List[ca.SX]:
+    half_distances = []
+    for i in range(point.size()[0]):
+        half_distances.append(
+            ca.fmax(
+                ca.fabs(point[i] - rectangle_center[i]) - rectangle_size[i] / 2,
+                0.0,
+            )
+        )
+    return half_distances
+
+
+def rectangle_to_point(
+    rectangle_center: ca.SX,
+    rectangle_size: ca.SX,
+    point: ca.SX,
+) -> ca.SX:
+    half_distances = cuboid_to_point_half_distances(
+        rectangle_center, rectangle_size, point
+    )
+    return ca.sqrt(half_distances[0] ** 2 + half_distances[1] ** 2)
+
+
+def rectangle_to_line(
+    rectangle_center: ca.SX,
+    rectangle_size: ca.SX,
+    line_start: ca.SX,
+    line_end: ca.SX,
+) -> ca.SX:
+    min_distance = ca.fmin(
+        rectangle_to_point(rectangle_center, rectangle_size, line_start),
+        rectangle_to_point(rectangle_center, rectangle_size, line_end),
+    )
+    for i in [-1, 1]:
+        for j in [-1, 1]:
+            index = [i, j]
+            corner_transform = rectangle_size / 2 * index
+            corner = rectangle_center + corner_transform
+            min_distance = ca.fmin(
+                min_distance, point_to_line(corner, line_start, line_end)
+            )
+    return min_distance
+
+
+def cuboid_to_point(
+    cuboid_center: ca.SX,
+    cuboid_size: ca.SX,
+    point: ca.SX,
+) -> ca.SX:
+    half_distances = cuboid_to_point_half_distances(
+        cuboid_center, cuboid_size, point
+    )
+    return ca.sqrt(
+        half_distances[0] ** 2 + half_distances[1] ** 2 + half_distances[2] ** 2
+    )
+
+
+def edge_of_cuboid(
+    cuboid_center: ca.SX, cuboid_size: ca.SX, index: int
+) -> ca.SX:
+    edges = [
+        [[-1, -1, -1], [1, -1, -1]],
+        [[-1, -1, -1], [-1, 1, -1]],
+        [[-1, -1, -1], [-1, -1, 1]],
+        [[-1, -1, 1], [1, -1, 1]],
+        [[-1, -1, 1], [-1, 1, 1]],
+        [[-1, 1, -1], [1, 1, -1]],
+        [[-1, 1, -1], [-1, 1, 1]],
+        [[1, -1, -1], [1, 1, -1]],
+        [[1, -1, -1], [1, -1, 1]],
+        [[1, 1, 1], [-1, 1, 1]],
+        [[1, 1, 1], [1, -1, 1]],
+        [[1, 1, 1], [1, 1, -1]],
+    ]
+    edge_start: ca.SX = cuboid_center + cuboid_size / 2 * edges[index][0]
+    edge_end: ca.SX = cuboid_center + cuboid_size / 2 * edges[index][1]
+    return ca.vertcat(edge_start, edge_end)
+
+
+def cuboid_to_line(
+    cuboid_center: ca.SX,
+    cuboid_size: ca.SX,
+    line_start: ca.SX,
+    line_end: ca.SX,
+) -> ca.SX:
+    distance = ca.fmin(
+        cuboid_to_point(cuboid_center, cuboid_size, line_start),
+        cuboid_to_point(cuboid_center, cuboid_size, line_end),
+    )
```

Comment:
@saraybakker1 That was the only thing missing in your formula. It covers the case that the end of the line segment is in front of one of the side faces. 
I copied your initial implementation for the edge-to-line distance to cover the other cases.